mempool/src/pool/tests.rs

```diff
@@ -0,0 +1,1787 @@
+        .outpoint
+        .clone();
+
+    let outpoint_source_id =
+        OutPointSourceId::from(*outpoint.tx_id().get_tx_id().expect("Not Coinbase"));
+
+    let input = TxInput::new(
+        outpoint_source_id,
+        0,
+        InputWitness::NoSignature(Some(DUMMY_WITNESS_MSG.to_vec())),
+    );
+    let flags = 1;
+    let locktime = 0;
+    let original = tx_spend_input(&mempool, input.clone(), original_fee, flags, locktime)
+        .expect("should be able to spend here");
+    let original_id = original.get_id();
+    eprintln!("created a tx with fee {:?}", mempool.try_get_fee(&original));
+    mempool.add_transaction(original)?;
+
+    let flags = 0;
+    let replacement = tx_spend_input(&mempool, input, replacement_fee, flags, locktime)
+        .expect("should be able to spend here");
+    eprintln!(
+        "created a replacement with fee {:?}",
+        mempool.try_get_fee(&replacement)
+    );
+    mempool.add_transaction(replacement)?;
+    assert!(!mempool.contains_transaction(&original_id));
+
+    Ok(())
+}
+
+#[test]
+fn try_replace_irreplaceable() -> anyhow::Result<()> {
+    let mut mempool = setup();
+    let outpoint = mempool
+        .available_outpoints(true)
+        .iter()
+        .next()
+        .expect("there should be an outpoint since setup creates the genesis transaction")
+        .outpoint
+        .clone();
+
+    let outpoint_source_id =
+        OutPointSourceId::from(*outpoint.tx_id().get_tx_id().expect("Not Coinbase"));
+
+    let input = TxInput::new(
+        outpoint_source_id,
+        0,
+        InputWitness::NoSignature(Some(DUMMY_WITNESS_MSG.to_vec())),
+    );
+    let flags = 0;
+    let locktime = 0;
+    let original_fee = Amount::from_atoms(get_relay_fee_from_tx_size(TX_SPEND_INPUT_SIZE));
+    let original = tx_spend_input(&mempool, input.clone(), original_fee, flags, locktime)
+        .expect("should be able to spend here");
+    let original_id = original.get_id();
+    mempool.add_transaction(original)?;
+
+    let flags = 0;
+    let replacement_fee = (original_fee + Amount::from_atoms(1000)).unwrap();
+    let replacement = tx_spend_input(&mempool, input, replacement_fee, flags, locktime)
+        .expect("should be able to spend here");
+    assert!(matches!(
+        mempool.add_transaction(replacement.clone()),
+        Err(Error::TxValidationError(
+            TxValidationError::ConflictWithIrreplaceableTransaction
+        ))
+    ));
+
+    mempool.drop_transaction(&original_id);
+    mempool.add_transaction(replacement)?;
+
+    Ok(())
+}
+
+#[test]
+fn tx_replace() -> anyhow::Result<()> {
+    let relay_fee = get_relay_fee_from_tx_size(TX_SPEND_INPUT_SIZE);
+    let replacement_fee = Amount::from_atoms(relay_fee + 100);
+    test_replace_tx(Amount::from_atoms(100), replacement_fee)?;
+    let res = test_replace_tx(Amount::from_atoms(300), replacement_fee);
+    assert!(matches!(
```

Comment:
I am looking forward for `assert_matches!`

Reply:
Me too :-D.